strongbox-commons/src/main/java/org/carlspring/strongbox/config/YamlPropertyPathResolverConfig.java

```diff
@@ -0,0 +1,82 @@
+package org.carlspring.strongbox.config;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.util.StringUtils;
+
+/**
+ * Configures a {@link }PropertiesPathResolver} under the name "propsPath" - it can then be used in SpEL
+ * expressions to resolve properties lookup path's such as {@code @Value("#{propsPath.resolve('customPropName','defaultPropPath')}")}.
+ *
+ * @author cbono
+ */
+@Configuration
+class YamlPropertyPathResolverConfig
+{
+    @Bean(name = "propsPath")
+    PropertiesPathResolver propertiesPathResolver()
+    {
+        return new PropertiesPathResolver();
+    }
+
+    /**
+     * Resolves a final path to use to lookup a properties resource.
+     */
+    static class PropertiesPathResolver
+    {
+        static final String PREFIX_OVERRIDE_PROPERTY = "strongbox.props.default-location-prefix";
+
+        private static final Logger logger = LoggerFactory.getLogger(PropertiesPathResolver.class);
+
+        @Inject
+        private Environment environment;
+
+        /**
+         * Resolves a final path to use to lookup a properties resource.
+         * <p>
+         * If there is a user-specified path set in the {@code customPathPropertyName} property then it will used and
+         * prefixed with 'file:' (unless it is already prefixed with 'classpath:' - which should be a rare case).
+         * <p>
+         * Otherwise, the specified {@code defaultPath} will be used and will be prefixed with
+         * {@code 'file:${strongbox.home}'} (unless there is a default path prefix override specified in the
+         * {@link #PREFIX_OVERRIDE_PROPERTY} property).
+         *
+         * @param customPathPropertyName the name of the property that may hold a user specified path
+         * @param defaultPath default path to use if no user specified path is set in the custom path property
+         * @return final path to use to lookup the YAML resource - including the proper prefix ('file:' or 'classpath:')
```

Comment:
> @return final path to use to lookup the YAML resource - including the proper prefix ('file:' or 'classpath:')

Still YAML related

Reply:
Ah yeh - missed these files. Will update in a bit.